<!DOCTYPE html>
<html lang="en-gb">
  
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <link rel="stylesheet" type="text/css" href="attempt1.new.css">

<head>
    <title>A study of ...</title>
    <center><h1>A study of...</h1></center>
    <hr>
</head>
<body>

<button class="tablink" onclick="openCity('Home', this, 'red')" id="defaultOpen">Home</button>
<button class="tablink" onclick="openCity('Summary', this, 'blue')">Summary</button>
<button class="tablink" onclick="openCity('Re-analysis', this, 'orange')">Re-analysis</button>
<button class="tablink" onclick="openCity('PigeonPigs', this, 'green')">PigeonPigs</button>


<div id="Home" class="tabcontent">
  <h1>A study of...</h1>
  		<p>A re-analysis and a new analysis of  <a href="https://www.pnas.org/content/112/23/7285">'A survey of human brain transcriptome diversity at the single cell level'</a>  by Darmanis <i>et al.</i> (2015)</p>
<br>
<p>Download our full report <a href="./images/SRP.pdf" download>here!</a></p>
<br>
<p>We put all our code on GitHub! Find it <a href="https://github.com/rpg18/SRP">here!</a></p>
<br>
<h2>PigeonPigs</h2>
<p>Your favourite pipeline!</p>
    <img src="./images/PigeonPigs.png" width="600" height="1000">


</div>


<div id="Summary" class="tabcontent">
  <h1>Summary</h1>
	<h3>Single-cell technologies provide novel insight into individual cell types as well as evolution and cell life cycles. However, with it comes a large quantity of data to analyse. With a lack of standardised method, there remain many approaches to this. Darmanis <i>et al.</i> (2015) use this new technique to analyse the cells within the brain. <br><br>
This website presents our attempt to re-analyse their data using the methodology they provide as well as creating a new and improved pipeline - PigeonPigs.
We were able to reproduce the general conclusions of Darmanis <i>et al.</i> (2015), but not an exact replication.<br><br>
The novel scRNAseq data analysis procedure PigeonPigs was able to reproduce much of their classification of cells into main brain cell types and discovered some new features in the dataset.</h3> 
</div>


<div id="Re-analysis" class="tabcontent">
		<h3>Re-analysis</h3>
    <li><a href="./pages/MHC.html">Expression of MHC I pathway genes in neuronal cells</a></li>
<br>
		<li><a href="./pages/MST.html">Minimum spanning tree for all neuronal cells</a></li>
<br>
		<li><a href="./pages/izzie_tsne.html">Clustering by tSNE</a></li>
<br>	
		<li><a href="./pages/pca_neuron.html">PCA neurons</a></li>
<br>
<p>It should be noted that where we reference cell type, the cell type was assigned by Darmanis <i>et al.</i> (2015)</p>
<br>
<p>Flowchart of our re-analysis pipeline</p>
<div style="display: inline-block"><img src="./images/Re1.png" width="600" height="900"></div>
<div style="display: inline-block"><img src="./images/Re2.png" width="600" height="500"></div>
</div>


<div id="PigeonPigs" class="tabcontent">
		<h3>PigeonPigs</h3>
    <li><a href="./pages/tsne_pca_graphs.html">3D tSNE and PCA visualisation</a></li>
		<li><a href="./pages/new_comparison.html">Clustering by SC3</a></li>
		<li><a href="./pages/genes_db.html">Gene Database</a></li>	
		<li><a href="./pages/diffusion_map.html">2D visualisation by gene</a></li>	
<br>
<p>It should be noted that where we reference cell type, the cell type was assigned by Darmanis <i>et al.</i> (2015)</p>
<br>
<p><a href="./images/FinalObject" download>Download our SingleCellExperiment R object here!</a></p>
<p><a href="./images/Merged_out_mmquant.csv" download>Download our gene count matrix here!</a></p>
<br>
<br>
<p>Heatmap of Marker genes for 9 SC3 clusters</p>
    <img src="./images/noFilt_markers_k9_clusters_9_age.png" width="1000" height="1000">
<p>Heatmap plot generated by sc3_plot_markers function with k = 9 from SC3, using the unfiltered matrix from PigeonPigs pipeline: The plot shows a maximum of ten marker genes per cluster, according to the total number of clusters. Here 9 clusters are selected. The log2 outlier score  is represented in a green scale: from 0 to 4. These scores were calculated with the sc3_calc_biology function, which identifies outlier cells in the dataset. The cell type classification from Darmaris <i>et al.</i>, (2015) is also shown. The marker genes for each cluster are represented as rows.</p>
<br>
<p>Flowchart of PigeonPigs pipeline</p>
    <img src="./images/PigeonPigs.png" width="600" height="1000">		
</div>

    
<script>
function openCity(cityName,elmnt,color) {
  var i, tabcontent, tablinks;
  tabcontent = document.getElementsByClassName("tabcontent");
  for (i = 0; i < tabcontent.length; i++) {
    tabcontent[i].style.display = "none";
  }
  tablinks = document.getElementsByClassName("tablink");
  for (i = 0; i < tablinks.length; i++) {
    tablinks[i].style.backgroundColor = "";
  }
  document.getElementById(cityName).style.display = "block";
  elmnt.style.backgroundColor = color;

}
// Get the element with id="defaultOpen" and click on it
document.getElementById("defaultOpen").click();
</script>
    
<footer class="w3-container w3-padding-64 w3-center w3-black w3-xlarge">
  <p class="w3-medium">
    Powered by Graham, Raquel and Izzie
  </p>
</footer>

</body>
</html>



